### ISSUE TITLE
graphql_fields() fails if a type has a list of union type

### ISSUE THREAD

**Issue by kyle-kc** [denisart/graphql-query#40]
**Describe the bug**
If a subclass of `GraphQLQueryBaseModel` has a field with a type as a list of union, calling `graphql_fields()` will cause an error.

```
AttributeError: '_SpecialForm' object has no attribute 'model_fields'
```

**To Reproduce**
Create the following classes:

```
class ClassA:
    x: int

class ClassB:
    y: int

class ClassC:
    z: List[Union[ClassA, ClassB]]
```

Then call `ClassC.graphql_fields()`

**Expected behavior**
This should work and produce the same output as if `z` were simply `Union[ClassA, ClassB]` (as list queries look no different than singular queries in GraphQL).

**Additional context**
As a workaround, one can simply set the type of `z` to `Union[ClassA, ClassB]` to produce the correct query when rendered, but this causes other issues if these classes are used for deserialization or for other purposes, as the type of `z` will be incorrect at runtime.

**Comment by denisart** [comment]
Hello, thanks for using graphql-query!

It is a bug. I fixed it in this https://github.com/denisart/graphql-query/pull/41

**Comment by denisart** [comment]
Available in version 1.4.0
